### PR TITLE
Updates to accommodate CI/CD

### PR DESCRIPTION
**Pull request by galtm**
# Committer Notes

{Please provide a brief description of what this PR accomplishes. Be sure to reference any issues addressed. If the PR is a work-in-progress submitted for early review, please include [WIP] at the beginning of the title or mark the PR as DRAFT.}

This PR imitates changes in xslt3-functions PR#7 to make XSpec tests compatible with CI/CD.

- CI/CD puts compiled XSpec file somewhere else, so define ov:service variable more robustly so that uuid-value.txt will be found regardless of where compiled XSpec file is.

- Java UUID class is not found in CI/CD, so conditionalize tests that use it. Running XSpec in Oxygen finds the class and verifies its results (same as before this change).

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/usnistgov/OSCAL/blob/main/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Do all automated CI/CD checks pass?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

(For reviewers: [The wiki has guidance](https://github.com/usnistgov/OSCAL/wiki/Issue-Review) on code review and overall issue review for completeness.)

### Changes to Core Features:

N/A because this is not a feature change.

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.

### CODE REVIEW

**Comment by iMichaela** [comment]
@galtm - The path to the "missing file" was addressed by your change. There are other tests failing. Majority, due to another relative path not well defined (details below, based on local execution):
```
/Library/Developer/CommandLineTools/usr/bin/make -C ../src/utils/resolver-pipeline test
TEST_DIR=./testing/xspec \
                XSPEC_COMMAND=../../../build/metaschema-xslt/support/xspec/bin/xspec.sh \
                SAXON_CP="/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/java_modules/*" \
                ./testing/test.sh
xspec_suite_path,status
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/4_modified/modify.xspec,failed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/2_metadata/random-util.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/2_metadata/uuid-method-choice.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/2_metadata/metadata.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/2_metadata/version-util.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/5_finished/finish.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/1_selected/glob-rewrite.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/1_selected/select-rlink.xspec,failed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/1_selected/select.xspec,failed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/1_selected/select-mapping-controls.xspec,failed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/3_merged/merge.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/3_merged/merge-as-is.xspec,passed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/3_merged/merge-custom.xspec,failed
/Users/miorga/Desktop/ALL_PROJECTS/GH/OSCAL/src/utils/resolver-pipeline/testing/3_merged/merge-combine.xspec,passed
```
Here is first error:
```
Type error at char 69 in expression in xsl:sequence/@select on line 18885 column 191 of modify-compiled.xsl:
  XPTY0004  An empty sequence is not allowed as the first argument of
  patches-to-id-targeting-ancestor()
at template scenario7-scenario7 on line 18705 column 83 of modify-compiled.xsl:
     invoked by xsl:call-template at file:/Users/miorga//OSCAL/src/utils/resolver-pipeline/./testing/xspec/modify-compiled.xsl#15819
at template scenario7 on line 15789 column 83 of modify-compiled.xsl:
     invoked by xsl:for-each at file:/Users/miorga//OSCAL/src/utils/resolver-pipeline/./testing/xspec/modify-compiled.xsl#15798
at template scenario7 on line 15789 column 83 of modify-compiled.xsl:
     invoked by xsl:call-template at file:/Users/miorga//OSCAL/src/utils/resolver-pipeline/./testing/xspec/modify-compiled.xsl#238
at template main on line 201 column 41 of modify-compiled.xsl:
     invoked by xsl:for-each at file:/Users/miorga//OSCAL/src/utils/resolver-pipeline/./testing/xspec/modify-compiled.xsl#217
at template main on line 201 column 41 of modify-compiled.xsl:
An empty sequence is not allowed as the first argument of patches-to-id-targeting-ancestor()
```
The other tests appear to fail due to the relative path I mentioned:
```
Error at char 9 in expression in xsl:sequence/@select on line 155 column 58 of oscal-profile-resolve-select.xsl:
  FODC0002  I/O error reported by XML parser processing
  file:/Users/miorga//specifications/profile-resolution/profile-resolution-examples/catalogs/xyz-tiny_catalog.xml. Caused by java.io.FileNotFoundException: /Users/miorga//specifications/profile-resolution/profile-resolution-examples/catalogs/xyz-tiny_catalog.xml (No such file or directory) 
```
Everywhere where the path is define as:
```
select="resolve-uri('../../../../../../specifications/profile-resolution/profile-resolution-examples')"
```
It should be:
```
select="resolve-uri(‘../../../../specifications/profile-resolution/profile-resolution-examples')"
```
since the project tree is:

OSCAL/src/specifications/profile-resolution/proile-resolution-examples/catalogs/xyz-tiny_catalog.xml
 OSCAL/src/utils/resolver-pipeline/testing/1_selected/*.xspec

File: src/utils/resolver-pipeline/testing/4_modified/modify.xspec [hunk 1]
```diff
@@ -1158,7 +1158,7 @@
         </x:scenario>
         <x:scenario label="Edge case: Empty $modifications param">
             <x:call function="oscal:patches-to-id-targeting-ancestor">
-                <x:param name="here" select="id('a1-stmt')">
+                <x:param name="here" select="//*[@id='a1-stmt']">
```
Comment (galtm) [inline src/utils/resolver-pipeline/testing/4_modified/modify.xspec:1161]: For the record: `id(...)` in this context was incorrect all along, so I wondered why the test passed in Oxygen but not on the command line. I believe it's a difference in the Saxon edition in use. Oxygen uses Saxon-EE in the default XSpec transformation scenario, but on the command line I was using Saxon-HE. When I force Oxygen to use Saxon-HE, I get the same error as in the command line run.

I speculate that Saxon-EE might have some optimization that makes the function call bypass data type checking of the first input argument. Perhaps Saxon-EE figures out that in the function, the `$alterations` variable is empty, so the return value of the function is empty regardless of the first input.

Comment (iMichaela) [reply src/utils/resolver-pipeline/testing/4_modified/modify.xspec:1161]: The CI/CD pipeline uses Saxon-HE 11.5. The latest version is 12.4, but @nikitawootten-nist indicated earlier attempts to upgrade to Saxon-HE 12.0, caused issues. I haven't tried the latest version. Which version of Saxon-HE have you used?

Comment (galtm) [reply src/utils/resolver-pipeline/testing/4_modified/modify.xspec:1161]: In my comment above, I used Saxon-EE 12.3 with Oxygen and Saxon-HE 12.3 outside Oxygen.

Just now, I tried the 6 XSpec files in this pull request with Saxon-HE 11.5 outside Oxygen (in a Windows shell using `xspec.bat`). Still no failures.

Comment (iMichaela) [reply src/utils/resolver-pipeline/testing/4_modified/modify.xspec:1161]: Thank you very much, @galtm ! I greatly appreciate your help, Amanda. I will run tomorrow all tests.

Comment (galtm) [reply src/utils/resolver-pipeline/testing/4_modified/modify.xspec:1161]: On the Checks tab of this PR, in the "Run all pipelines" section, I see evidence of these tests running (e.g., log includes "PENDING: (Needs confirmation about latest expected result) ..include-controls with parent controls (default choice)"), and I do not see failures. Line 1650  of that log says, "All test suites passed ✅"

**Comment by iMichaela** [comment]
Indeed. All tests passed. I will review one more time today and approve it.

File: src/utils/resolver-pipeline/testing/4_modified/modify.xspec (hunk 1 above)
Comment (wendellpiez) [reply src/utils/resolver-pipeline/testing/4_modified/modify.xspec:1161]: Functionality of `id()` can depend on a schema binding, which is one of the features offered by Saxon-PE/EE, available in oXygen but not in Saxon-HE. So I am not entirely surprised to see that wiggle.

**Review by wendellpiez — APPROVED** [review]
Beautiful work thanks @galtm

**Pull request merged**